coffee/base.py

```diff
@@ -1209,6 +1222,26 @@ def gencode(self, not_scope=False):
         return self.children[0].gencode()
 
 
+class Rank(tuple):
+
+    def __contains__(self, val):
```

Comment:
Fine, I do not mind having the interface cleanup (which will require corresponding changes in other components) in a separate PR after this one is merged.